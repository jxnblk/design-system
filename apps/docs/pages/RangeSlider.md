# RangeSlider

A single or multi-value range input based on [react-component-slider][rc-slider]

```.jsx
<RangeSlider
  min={0}
  max={100}
  defaultValue={[8, 64]}
/>
```

## Props

| Prop                                | Type              | Description                                                           |
| ----------------------------------- | ----------------- | --------------------------------------------------------------------- |
| `min`                               | number            | The minimum value of the slider                                       |
| `max`                               | number            | The maximum value of the slider                                       |
| `defaultValue`                      | array             | Set initial value of slider                                           |
| `value`                             | array             | Set current value of slider                                           |
| `onChange`                          | function          | triggered when value changes                                          |
| `ariaLabelGroupForHandles`          | array (strings)   | Set aria-label for each handle                                        |
| `ariaLabelledByGroupForHandles`     | array (strings)   | Set aria-labelledby for each handle                                   |
| `ariaValueTextFormattersForHandles` | array (functions) | Pass a function to each handle to calculate aria-valuetext from value |

For more props, see the [react-component-slider docs][rc-slider]

[rc-slider]: https://github.com/react-component/slider
