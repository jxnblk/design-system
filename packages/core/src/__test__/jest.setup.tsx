import React from 'react'
import { ThemeProvider } from 'styled-components'

import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import renderer from 'react-test-renderer'

import { createTheme } from '..'

global.rendererCreateWithTheme = (tree) =>
  renderer.create(<ThemeProvider theme={createTheme()}>{tree}</ThemeProvider>)

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
