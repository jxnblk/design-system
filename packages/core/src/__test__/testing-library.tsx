import { render, RenderResult } from '@testing-library/react'
import PropTypes from 'prop-types'
import React from 'react'

import { ThemeProvider } from '..'

const WithThemeProvider = ({ children }) => <ThemeProvider>{children}</ThemeProvider>
WithThemeProvider.propTypes = {
  children: PropTypes.node,
}

const customRender: (Element, any?) => RenderResult = (ui, options) =>
  render(ui, { wrapper: WithThemeProvider, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }

export const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
