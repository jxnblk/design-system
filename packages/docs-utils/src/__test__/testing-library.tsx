import { render, RenderResult } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'
import PropTypes from 'prop-types'
import React from 'react'

const WithThemeProvider = ({ children }) => <ThemeProvider>{children}</ThemeProvider>
WithThemeProvider.propTypes = {
  children: PropTypes.node,
}

// @ts-ignore
const customRender: (Element, any?) => RenderResult = (ui, options) =>
  render(ui, { wrapper: WithThemeProvider, ...options })

// re-export everything
export * from '@testing-library/react'
// override render method
export { customRender as render }
export const sleep = (ms: number) => new Promise((r) => setTimeout(r, ms))
