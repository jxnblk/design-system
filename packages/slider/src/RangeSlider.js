import PropTypes from 'prop-types'
import { space } from 'styled-system'

import Range from './rc-slider/Range'
import styleSlider from './styleSlider'

const RangeSlider = styleSlider(Range)

RangeSlider.defaultProps = {
  allowCross: false,
  color: 'primary',
}

RangeSlider.propTypes = {
  ...space.propTypes,
  color: PropTypes.string,
}

export default RangeSlider
