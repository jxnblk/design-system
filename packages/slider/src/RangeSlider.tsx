import React from 'react'
import PropTypes from 'prop-types'
import { space } from 'styled-system'
import { deprecatedPropType } from 'pcln-design-system'

import Range from 'rc-slider'
import styleSlider from './styleSlider'

const _RangeSlider = styleSlider(Range)

const RangeSlider = (props) => {
  return (
    <_RangeSlider
      {...{
        ariaLabelForHandle: props.ariaLabelGroupForHandles,
        ariaLabelledByForHandle: props.ariaLabelledByGroupForHandles,
        ariaValueTextFormatterForHandle: props.ariaValueTextFormatterGroupForHandles,
        ...props,
      }}
    />
  )
}

RangeSlider.defaultProps = {
  allowCross: false,
  color: 'primary',
  range: true,
}

RangeSlider.propTypes = {
  ...space.propTypes,
  color: PropTypes.string,
  ariaLabelGroupForHandles: deprecatedPropType('ariaLabelForHandle'),
  ariaLabelledByGroupForHandles: deprecatedPropType('ariaLabelledByForHandle'),
  ariaValueTextFormatterGroupForHandles: deprecatedPropType('ariaValueTextFormatterForHandle'),
}

export default RangeSlider
