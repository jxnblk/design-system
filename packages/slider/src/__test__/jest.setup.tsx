import React from 'react'
import { ThemeProvider } from 'pcln-design-system'

import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import renderer from 'react-test-renderer'

global.rendererCreateWithTheme = (tree) => renderer.create(<ThemeProvider>{tree}</ThemeProvider>)

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
