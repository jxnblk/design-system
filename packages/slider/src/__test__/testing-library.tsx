import type { RenderOptions, RenderResult } from '@testing-library/react'
import type { ReactElement } from 'react'

import { render } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
