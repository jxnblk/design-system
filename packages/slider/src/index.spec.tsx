import React from 'react'
import { render } from './__test__/testing-library'

import RangeSlider, { Slider } from '.'

describe('pcln-slider', () => {
  test('RangeSlider', () => {
    const { asFragment } = render(<RangeSlider />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('Slider', () => {
    const { asFragment } = render(<Slider />)
    expect(asFragment()).toMatchSnapshot()
  })
})
