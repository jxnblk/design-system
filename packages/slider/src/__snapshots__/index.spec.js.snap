// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pcln-slider RangeSlider 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  position: relative;
  height: 32px;
  padding-top: 12px;
  border-radius: 9999px;
  touch-action: none;
}

.c1 .rc-slider-rail,
.c1 .rc-slider-track {
  height: 8px;
}

.c1 .rc-slider-handle {
  width: 32px;
  height: 32px;
  margin-left: -16px;
  margin-top: -12px;
}

.c1 .rc-slider-rail {
  position: absolute;
  width: 100%;
  background-color: #f4f6f8;
  border-radius: 9999px;
}

.c1 .rc-slider-track {
  position: absolute;
  left: 0;
  border-radius: 9999px;
  background-color: #0068EF;
}

.c1 .rc-slider-handle {
  position: absolute;
  cursor: pointer;
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  border-radius: 9999px;
  border: solid 4px #0068EF;
  background-color: #fff;
  touch-action: pan-x;
}

.c1 .rc-slider-handle:focus {
  box-shadow: 0 0 0 2px #e8f2ff;
}

.c1.rc-slider-disabled {
  color: #c0cad5;
}

.c1.rc-slider-disabled .rc-slider-handle {
  box-shadow: none;
  cursor: default;
}

<div
    class="c0"
  >
    <div
      class="rc-slider c1"
    >
      <div
        class="rc-slider-rail"
      />
      <div
        class="rc-slider-track rc-slider-track-1"
        style="left: 0%; width: 0%;"
      />
      <div
        class="rc-slider-step"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="rc-slider-handle rc-slider-handle-1"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="rc-slider-handle rc-slider-handle-2"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
      <div
        class="rc-slider-mark"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`pcln-slider Slider 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  position: relative;
  height: 32px;
  padding-top: 12px;
  border-radius: 9999px;
  touch-action: none;
}

.c1 .rc-slider-rail,
.c1 .rc-slider-track {
  height: 8px;
}

.c1 .rc-slider-handle {
  width: 32px;
  height: 32px;
  margin-left: -16px;
  margin-top: -12px;
}

.c1 .rc-slider-rail {
  position: absolute;
  width: 100%;
  background-color: #f4f6f8;
  border-radius: 9999px;
}

.c1 .rc-slider-track {
  position: absolute;
  left: 0;
  border-radius: 9999px;
  background-color: #0068EF;
}

.c1 .rc-slider-handle {
  position: absolute;
  cursor: pointer;
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  border-radius: 9999px;
  border: solid 4px #0068EF;
  background-color: #fff;
  touch-action: pan-x;
}

.c1 .rc-slider-handle:focus {
  box-shadow: 0 0 0 2px #e8f2ff;
}

.c1.rc-slider-disabled {
  color: #c0cad5;
}

.c1.rc-slider-disabled .rc-slider-handle {
  box-shadow: none;
  cursor: default;
}

<div
    class="c0"
  >
    <div
      class="rc-slider c1"
    >
      <div
        class="rc-slider-rail"
      />
      <div
        class="rc-slider-track"
        style="left: 0%; width: 0%;"
      />
      <div
        class="rc-slider-step"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="rc-slider-handle"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
      <div
        class="rc-slider-mark"
      />
    </div>
  </div>
</DocumentFragment>
`;
